Planner/app/src/main/java/rdatu/android/cyscorpions/com/projectplanner/view/TimePickerDialog.java

```diff
@@ -0,0 +1,84 @@
+package rdatu.android.cyscorpions.com.projectplanner.view;
+
+import android.app.Activity;
+import android.app.Dialog;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.app.DialogFragment;
+import android.widget.TimePicker;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+
+/**
+ * Created by rayeldatu on 7/28/15.
+ */
+public class TimePickerDialog extends DialogFragment implements android.app.TimePickerDialog.OnTimeSetListener {
+
+    public static final String EXTRA_TIME = "time";
+    private String mStringTime;
+    private Callbacks mCallbacks;
+
+
+    public TimePickerDialog() {
+    }
+
+
+    public static TimePickerDialog newInstance(String time) {
+        Bundle args = new Bundle();
+        TimePickerDialog fragment = new TimePickerDialog();
+        args.putSerializable(EXTRA_TIME, time);
+        fragment.setArguments(args);
+
+        return fragment;
+    }
+
+    @Override
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
+        mCallbacks = (Callbacks) activity;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        mStringTime = getArguments().getString(EXTRA_TIME);
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(Bundle savedInstanceState) {
+        Date t = null;
+        SimpleDateFormat df = new SimpleDateFormat("kk:mm");
+        Calendar c = Calendar.getInstance();
+        try {
+            t = df.parse(mStringTime);
+        } catch (Exception e) {
+
+        }
+        c.setTime(t);
```

Comment:
If there's an exception during the parsing at line 56, this line of code will produce NullPointerException since t is still null